bin/new_element.dart

```diff
@@ -0,0 +1,195 @@
+
+  parser.addOption('output-dir', abbr: 'o', help: 'Output directory');
+  parser.addFlag('help', abbr: 'h');
+
+  var options, element;
+  try {
+    options = parser.parse(args);
+    if (options['help']) {
+      printUsage(parser);
+      return;
+    }
+    if (options.rest == null || options.rest.isEmpty) {
+      throw new FormatException('No element specified');
+    }
+    element = options.rest[0];
+    if (!_isPolymerElement(element)) {
+      throw new FormatException('Must specify polymer-element to create.\n'
+          'polymer-element must be all lowercase with at least 1 hyphen.');
+    }
+  } catch (e) {
+    print('$e\n');
+    printUsage(parser);
+    exitCode = 1;
+    return;
+  }
+
+  var outputDir, startDir;
+
+  var outputPath = options['output-dir'];
+
+  if (outputPath == null) {
+    if ((new File('pubspec.yaml')).existsSync()) {
+      print('When creating elements in root directory of package, '
+          '-o <dir> must be specified');
+      exitCode = 1;
+      return;
+    }
+    outputDir = (new Directory('.')).resolveSymbolicLinksSync();
+  } else {
+    var outputDirLocation = new Directory(outputPath);
+    if (!outputDirLocation.existsSync()) {
+      outputDirLocation.createSync(recursive: true);
+    }
+    outputDir = (new Directory(outputPath)).resolveSymbolicLinksSync();
+  }
+
+  var pubspecDir = _findDirWithFile(outputDir, 'pubspec.yaml');
+
+  if (pubspecDir == null) {
+    print('Could not find pubspec.yaml when walking up from $outputDir');
+    exitCode = 1;
+    return;
+  }
+
+  var length = path.split(pubspecDir).length;
+  var distanceToPackageRoot = path.split(outputDir).length - length;
+
+  // See dartbug.com/20076 for the algorithm used here.
+  if (distanceToPackageRoot > 0) {
+    if (path.split(outputDir)[length] == 'lib') {
+      distanceToPackageRoot++;
+    } else {
+      distanceToPackageRoot--;
+    }
+  }
+
+  try {
+    _createBoilerPlate(element, outputDir, distanceToPackageRoot);
+  } on Exception catch (e, t) {
+    print('Error creating files in $outputDir');
+    print('$e $t');
+    exitCode = 1;
+    return;
+  }
+
+  return;
+}
+
+String _findDirWithFile(String dir, String filename) {
+  while (!new File(path.join(dir, filename)).existsSync()) {
+    var parentDir = path.dirname(dir);
+    // If we reached root and failed to find it, bail.
+    if (parentDir == dir) return null;
+    dir = parentDir;
+  }
+  return dir;
+}
+
+bool _isPolymerElement(String element) {
+  return element.contains('-') && (element.toLowerCase() == element);
+}
+
+String _toCamelCase(String s) {
+  var segments = s.split('-');
+  return segments.map((s) => s[0].toUpperCase() + s.substring(1)).join('');
+}
+
+void _createBoilerPlate(
+    String element, String directory, int distanceToPackageRoot) {
+
+  var capitalizedName = _toCamelCase(element);
+  var underscoreName = element.replaceAll('-', '_');
+
+  String html = '''
+<dom-module id="$element">
+    <template>
```

Comment:
Sorry to keep nitpicking this but lets stick with either 4 or 2 space indent throughout. I don't have an opinion really on which, but it should be consistent. (template tag here is indented 4, rest is indented 2)